frontend/src/app/[locale]/(base)/organization/[id]/manage-users/page.tsx

```
@@ -14,19 +19,49 @@ export async function generateMetadata({
   params,
 }: {
   params: Promise<{ locale: string; id: string }>;
-}) {
-  const { locale } = await params;
+}): Promise<Metadata> {
+  const { locale, id } = await params;
   const t = await getTranslations({ locale });
-
+  let title = `${t("ManageUsers.pageTitle")}`;
+  try {
+    const session = await getSession();
+    if (!session?.token) {
+      throw new Error("not logged in");
+    }
+    const organizationDetails = await getOrganizationDetails(session.token, id);
+    title = `${t("ManageUsers.pageTitle")} - ${organizationDetails.sam_gov_entity.legal_business_name || ""}`;
+  } catch (error) {
+    console.error("Failed to render page title due to API error", error);
+    if (parseErrorStatus(error as ApiRequestError) === 404) {
+      return notFound();
+    }
```

What happens for non 404 errors, like 500?  If we just want it to fail silently here, maybe a comment to that effect?